runtime/lua/vim/secure.lua

```diff
@@ -86,21 +111,57 @@ function M.read(path)
     trust[fullpath] = hash
   end
 
+  write_trust(trust)
+
+  return contents
+end
+
+--- Update the trust status of file at path in the trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+--- @param path (string) Path to a file to update status for.
+--- @param allow (boolean) If true, set status to allow, otherwise set to deny.
+function M.trust(path, allow)
+  vim.validate({ path = { path, 's' } })
+  vim.validate({ allow = { allow, 'b' } })
+
+  local fullpath = vim.loop.fs_realpath(vim.fs.normalize(path))
+  if not fullpath then
+    return
+  end
```

Comment:
I think if the user is explicitly adding a file to the trust db then we want some kind of indication if the file does not exist. Doesn't need to be an error necessarily, but we could use the Lua pattern of returning multiple values: `true, nil` on success, and `false, msg` on failure (where `msg` is a string indicating what failed, in this case that the file doesn't exist).

Reply:
Resolved in 379075cf4297986fc8ac03175eb2e7fc2eb1062f